submissions/aymkin/OOP-frogger/app.js

```diff
@@ -0,0 +1,143 @@
+let scoreCount = 0;
+
+const playerStartValues = {
+	x: 202,
+	y: 405,
+	sprite: 'images/char-boy.png',
+};
+
+const enemyStartValues = {
+	x: 0,
+	y: [63, 146, 229],
+	sprite: 'images/enemy-bug.png',
+};
+
+const FIELD = {
+	column: 101,
+	row: 83,
+	leftBorder: 101,
+	rightBorder: 303,
+	topBorder: 0,
+	bottomBorder: 322,
+};
+
+const Creature = function(x, y, sprite) {
+	this.x = x;
+	this.y = y;
+	this.sprite = sprite;
+};
+
+Creature.prototype.render = function() {
+	ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+const Enemy = function(x, y, sprite, gamer) {
+	Creature.call(this, x, y, sprite);
+	this.speed = this.getRandSpeed();
+	this.gamer = gamer;
+};
+
+Enemy.prototype = Object.create(Creature.prototype);
+
+Enemy.prototype.getRandSpeed = function(min = 75, max = 200) {
+	return Math.floor(Math.random() * (max - min + 1)) + min;
+};
+
+Enemy.prototype.update = function(dt) {
+	this.x = this.x + this.speed * dt;
+	if (this.x >= FIELD.rightBorder + FIELD.column * 2) {
+		this.x = -101;
+		this.speed = this.getRandSpeed();
+	}
+	this.isCollision();
+};
+
+Enemy.prototype.isCollision = function() {
+	if (
+		this.x + FIELD.column >= this.gamer.x &&
+		this.x <= this.gamer.x + FIELD.column &&
+		this.y + 10 == this.gamer.y
+	) {
+		this.gamer.loosePoints();
+		this.gamer.beginAgain();
+	}
+};
+
+const Player = function(x, y, sprite) {
+	Creature.call(this, x, y, sprite);
+};
+
+Player.prototype = Object.create(Creature.prototype);
+
+Player.prototype.update = function() {
+	this.setScore();
+};
+
+Player.prototype.beginAgain = function() {
+	this.x = playerStartValues.x;
+	this.y = playerStartValues.y;
+};
+
+Player.prototype.handleInput = function(key) {
+	switch (key) {
+		case 'left':
+			if (this.x >= FIELD.leftBorder) this.x -= FIELD.column;
+			break;
+		case 'right':
+			if (this.x <= FIELD.rightBorder) this.x += FIELD.column;
+			break;
+		case 'up':
+			if (this.y >= FIELD.topBorder) this.y -= FIELD.row;
+			break;
+		case 'down':
+			if (this.y <= FIELD.bottomBorder) this.y += FIELD.row;
+			break;
+	}
+};
+
+Player.prototype.setScore = function() {
+	if (this.y < 0) {
+		scoreCount++;
+		if (scoreCount < 3) {
```

Comment:
I guess, here it would be more clear if we use named constant like maxRounds or smth like that.